shared/src/types/types.ts

```diff
@@ -223,3 +225,11 @@ export interface AnalysisProfile {
   labelSelector: string;
   readOnly?: boolean;
 }
+export interface ChatModelPair {
+  streamingModel: BaseChatModel;
```

Comment:
some models support tools but only in non-streaming mode. This makes it so that our tool calling is still superior in such cases but we lose the streaming which is a tradeoff for better quality responses.